### PR TITLE
Add PaddedLayout for laying out objects with padding all around

### PR DESCRIPTION
**Pull request by stuartmscott**
### Description:
Adds a new layout which lays child objects with theme.Padding() around them.

### Checklist:

- [x] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

#### Where applicable:

- [x] Public APIs match existing style.
- [ ] Any breaking changes have a deprecation path or have been discussed.

### CODE REVIEW

File: layout/paddedlayout.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
+// Package layout defines the various layouts available to Fyne apps
```
Comment (andydotxyz) [inline layout/paddedlayout.go:1]: I don't think the package comments should be duplicated - only appearing in one file in the package.

File: layout/paddedlayout_test.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	"fyne.io/fyne/layout"
+	"fyne.io/fyne/theme"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPaddedLayout(t *testing.T) {
+	size := fyne.NewSize(100, 100)
+
+	obj := canvas.NewRectangle(color.NRGBA{0, 0, 0, 0})
```
Comment (andydotxyz) [inline layout/paddedlayout_test.go:18]: why not use color.Black?

File: layout/paddedlayout_test.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	container.Resize(size)
+
+	layout.NewPaddedLayout().Layout(container.Objects, size)
+
+	assert.Equal(t, obj.Size().Width, size.Width-2*theme.Padding())
+	assert.Equal(t, obj.Size().Height, size.Height-2*theme.Padding())
+}
+
+func TestPaddedLayout_MinSize(t *testing.T) {
+	text := canvas.NewText("FooBar", color.NRGBA{0, 0xff, 0, 0})
```
Comment (andydotxyz) [inline layout/paddedlayout_test.go:31]: if we're not going to see it I'd be tempted to use color.Black or color.White...

**Review by andydotxyz — COMMENTED** [review]
Looks great, but not sure about the comment dupe or the long hand colour selection.
Not going to say it as "request changes", interested in what you think.

File: layout/paddedlayout_test.go (hunk 3 above)
Comment (stuartmscott) [reply layout/paddedlayout_test.go]: fixed

File: layout/paddedlayout_test.go (hunk 2 above)
Comment (stuartmscott) [reply layout/paddedlayout_test.go]: fixed

File: layout/paddedlayout.go (hunk 1 above)
Comment (stuartmscott) [reply layout/paddedlayout.go]: fixed

**Comment by stuartmscott** [comment]
I completely agree with you - artifacts of copy/pasta - fixed now

**Review by andydotxyz — APPROVED** [review]

**Comment by stuartmscott** [comment]
Oh Travis, why do you hate me so?

```
--- FAIL: TestButton_ChangeTheme (0.21s)
    util.go:47: 
        	Error Trace:	util.go:47
        	            				util.go:32
        	            				button_test.go:264
        	            				util.go:184
        	            				button_test.go:260
        	Error:      	Not equal: 
        	Test:       	TestButton_ChangeTheme
        	Messages:   	Image did not match master. Actual image written to file:///home/travis/gopath/src/fyne.io/fyne/widget/testdata/failed/button_theme_changed.png.
```

**Pull request merged**